### PR TITLE
[Distributed]Add loss nan/inf checker

### PR DESCRIPTION
**Pull request by ForFishes**
### PR types

Breaking changes
### PR changes

Others
### Description

[Distributed]Add loss nan/inf checker

### CODE REVIEW

File: paddlenlp/trainer/trainer.py [hunk 1]
```diff
@@ -995,6 +995,10 @@ def _inner_training_loop(
                 else:
                     tr_loss_step = self.training_step(model, inputs)
 
+                if self.amp_dtype != "float16":
```
Comment (ZHUI) [inline paddlenlp/trainer/trainer.py]: ci 报错了，需要先判断， ` if args.fp16 `

Comment (ForFishes) [reply paddlenlp/trainer/trainer.py]: ok, i will fix it
